posix/posix.c

```diff
@@ -1576,22 +1553,21 @@ int posix_fcntl(int fd, unsigned int cmd, char *ustack)
 #define SIOCDELRT   _IOC(IOC_IN, 'S', 0x45, sizeof(struct rtentry))
 
 
-void ioctl_pack(msg_t *msg, unsigned long request, void *data, id_t id)
+static void ioctl_pack(msg_t *msg, unsigned long request, void *data, oid_t oid)
 {
 	size_t size = IOCPARM_LEN(request);
 	ioctl_in_t *ioctl = (ioctl_in_t *)msg->i.raw;
 	struct ifconf *ifc;
 	struct rtentry *rt;
 
+	msg->oid = oid;
```

Comment:
Potential memcpy being emitted
```suggestion
	hal_memcpy(&msg->oid, &oid, sizeof(oid));
```